modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -4823,7 +4823,11 @@ public CodegenParameter fromRequestBody(RequestBody body, Set<String> imports, S
         }
         schema = ModelUtils.getReferencedSchema(this.openAPI, schema);
 
-        if (ModelUtils.isMapSchema(schema)) {
+        boolean useAliasReferenceName = ModelUtils.isGenerateAliasAsModel() && StringUtils.isNotBlank(name) &&
+          (ModelUtils.isMapSchema(schema) || ModelUtils.isArraySchema(schema)) &&
+          (schema.getProperties() == null || schema.getProperties().isEmpty());
+
```

Comment:
@bodograumann what if the alias schema is a primitive like string/integer etc?
Why not allow those?
This update currently limits it to mapSchemas or ArraySchemas.

Reply:
Why aren't we using model.isAlias?

Reply:
Investigation into this led to this alternate PR:
https://github.com/OpenAPITools/openapi-generator/pull/4597
@bodograumann can you fully describe the models that you need adjusted?
It looks like they are not the primitive non-object non-array models that I am writing about.

Reply:
@bodograumann whaf does a non primitive non-object non-array look like? Can you paste in a spec sample of one?

Reply:
@bodograumann this update looks good but it is a bit difficult to understand the way you wrote it because for array and map isAlias use cases we are not going down the isMapSchema and isArraySchema routes.

Can you please update your PR to something like the following:
```
privat void addBodyModelSchema(blah...) {
  // this function adds a model schema to a request body
}

public CodegenParameter fromRequestBody(...) {
  if (ModelUtils.isMapSchema(schema) {
    if (ModelUtils.isGenerateAliasAsModel() && StringUtils.isNotBlank(name)) {
      addBodyModelSchema(blah..)
    } else {
      // non-alias model code here
    } 
  } else if (ModelUtils.isArraySchema(schema) {
    if (ModelUtils.isGenerateAliasAsModel() && StringUtils.isNotBlank(name)) {
      addBodyModelSchema(blah..)
    } else {
      // non-alias model code here
    } 
  } else if (ModelUtils.isObjectSchema(schema) {
    addBodyModelSchema(blah..)
  }
}
```
That way, it is very clear to the maintainers that
- we are using an isAlias feature
- we only use it in array schemas + map schemas
- and that we are using it to link up to an existing model